src/Isotope.cpp

```diff
@@ -0,0 +1,607 @@
+ * @return The value of the fission spectrum
+ * @param group The index of the group 
+ */
+FP_PRECISION Isotope::getChiByGroup(int group) {   
+  if (_chi == NULL)
+    log_printf(ERROR, "Unable to return isotope %d's fission "
+                      "spectrum since it has not yet been set", _uid);
+                      
+  if (group < 1 || group > _num_groups)
+    log_printf(ERROR, "Unable to return isotope %d's fission spectrum "
+                      " for group %d, since it only has %d groups.",
+                      _uid, group, _num_groups);
+       
+  return _chi[group-1];
+}  
+
+
+/**
+ * @brief Set the number of energy groups for this isotope.
+ * @param num_groups the number of energy groups.
+ */
+void Isotope::setNumEnergyGroups(const int num_groups) {
+
+  if (num_groups < 0)
+    log_printf(ERROR, "Unable to set the number of energy groups for "
+                 "isotope %d to %d", _id, _num_groups);
+
+  _num_groups = num_groups;
+
+  if (_sigma_t != NULL)
+    delete [] _sigma_t;
+
+  if (_sigma_a != NULL)
+    delete [] _sigma_a;
+
+  if (_sigma_s != NULL)
+    delete [] _sigma_s;
+  
+  if (_sigma_f != NULL)
+    delete [] _sigma_f;
+  
+  if (_nu_sigma_f != NULL)
+    delete [] _nu_sigma_f;
+  
+  if (_chi != NULL)
+    delete [] _chi;
+
+
+  _sigma_t = new FP_PRECISION[_num_groups];
+  _sigma_a = new FP_PRECISION[_num_groups];
+  _sigma_s = new FP_PRECISION[_num_groups*_num_groups];
+  _sigma_f = new FP_PRECISION[_num_groups];
+  _nu_sigma_f = new FP_PRECISION[_num_groups];
+  _chi = new FP_PRECISION[_num_groups];
+  
+
+  /* Assign the null vector to each data array */
+  memset(_sigma_t, 0.0, sizeof(FP_PRECISION) * _num_groups);
+  memset(_sigma_a, 0.0, sizeof(FP_PRECISION) * _num_groups);
+  memset(_sigma_s, 0.0, sizeof(FP_PRECISION) * _num_groups*_num_groups);
+  memset(_sigma_f, 0.0, sizeof(FP_PRECISION) * _num_groups);
+  memset(_nu_sigma_f, 0.0, sizeof(FP_PRECISION) * _num_groups);
+  memset(_chi, 0.0, sizeof(FP_PRECISION) * _num_groups);
+}
+
+
+/**
+ * @brief Set the isotope's array of total cross-sections.
+ * @param xs the array of total cross-sections
+ * @param num_groups the number of energy groups
+ */
+void Isotope::setSigmaT(double* xs, int num_groups) {
+
+  if (_num_groups != num_groups)
+    log_printf(ERROR, "Unable to set sigma_t with %d groups for isotope "
+       "%d which contains %d energy groups", num_groups, _uid,
+       _num_groups);
+
+  
+
+  for (int i=0; i < _num_groups; i++) 
+    _sigma_t[i] = FP_PRECISION(xs[i]);
+}
+
+
+/**
+ * @brief Set the isotope's total cross-section for some energy group.
+ * @param xs the total cross-section (\f$ \Sigma_t [cm^1] \f$)
+ * @param group the energy group
+ */
+void Isotope::setSigmaTByGroup(double xs, int group) {
+
+  if (group <= 0 || group > _num_groups)
+    log_printf(ERROR, "Unable to set sigma_t for group %d for isotope "
+              "%d which contains %d energy groups", group, _uid, _num_groups);
+
+  _sigma_t[group-1] = xs;
```

Comment:
And if you were simply copying from my original ``Material`` class implementation, then I apologize for causing a fuss :-)